h2/src/main/org/h2/engine/Session.java

```diff
@@ -173,6 +174,7 @@ public boolean equals(Object obj) {
     private final ValueTimestampTimeZone sessionStart;
     private ValueTimestampTimeZone transactionStart;
     private ValueTimestampTimeZone currentCommandStart;
+    private ValueTimestampTimeZone sleepStateSince;
```

Comment:
Maybe I'm wrong, but it looks like there is no need for two fields, a single shared filed may be used to store timestamp of last change. Session knows its state (see `Session.state`), so current meaning of the field can be easily determined.

Reply:
Agreed, maybe we rename the field to currentCommandStartOrEnd to indicate this dual use of the value?

Reply:
Just use something like `commandStartOrEnd`.

Reply:
Code now shares the field commandStartOrEnd for storing the timestamp